documentation/specs/Toast.md

```diff
@@ -0,0 +1,174 @@
+# `Toast` Component Specification
+
+## Overview
+
+The `Toast` component displays a brief non-disruptive message to the user as a result of an action taken. The displayed message is accompanied by an associated `status` icon and cannot be manually dismissed. **Not** to be used to display critical information such as a system failure, see the `Alert` component for such behavior.
+
+### Use Cases
+
+- Inform user of a completed action such as an account creation.
+- Inform user that an asynchronous process has been triggered based on their action.
+
+### Features
+
+- ARIA `role="status"`
```

Comment:
react-aria has a `useToast` hook, but seems like its not the aria pattern we want for our toast and might be tricky as to where we draw the line for whats within this library vs what is left up to the consumer https://react-spectrum.adobe.com/react-aria/useToast.html

Reply:
i do think we will want to utilize `useToast` both here and for the `Alert` component with some minor configuration differences between the two. it handles a lot of things that we will want to handle in Easy UI, such as rendering it into a labeled region, automatically focusing in and out, attaching accessibility labels, state management, and timers.

from what i understand, the aspect that we would need to adjust for in our `Toast` component would be setting the `role="status"` on the toast element directly, overriding the `alert` value that [comes from Aria](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/toast/src/useToast.ts#L91). from what i can tell, the rest of it would be the same.

let me know if i'm misunderstanding anything related to the accessibility/Aria situation.

Reply:
nope, youre good! i updated to use `react-aria` and `react-stately`. my initial hesitation for using these was for our use of the `Toast`, i wasn't sure if keyboard navigation was a necessity or would cause harm seeing as we wouldn't want to render anything interactive for an element that per our spec, needs to time out; but i think i had a misunderstanding initially about the purpose of the region landmark. for `react-stately`, it initially looked like we'd have to do some heavy configuration, but it ended up being not too bad.